sdk/go/common/workspace/plugins.go

```diff
@@ -353,14 +357,54 @@ func newGithubSource(url *url.URL, name string, kind PluginKind) (*githubSource,
 }
 
 func (source *githubSource) newHTTPRequest(url, accept string) (*http.Request, error) {
-	req, err := buildHTTPRequest(url, fmt.Sprintf("token %s", source.token))
+	var token string
+	if source.token != "" {
+		token = fmt.Sprintf("token %s", source.token)
+	}
+	req, err := buildHTTPRequest(url, token)
 	if err != nil {
 		return nil, err
 	}
 	req.Header.Set("Accept", accept)
 	return req, nil
 }
 
+func (source *githubSource) retryHTTP(
+	getHTTPResponse func(*http.Request) (io.ReadCloser, int64, error),
+	req *http.Request) (io.ReadCloser, int64, error) {
+
+	delay := 80 * time.Millisecond
+	for attempt := 0; ; attempt++ {
+		resp, length, err := getHTTPResponse(req)
+		if err == nil {
+			return resp, length, nil
+		}
+
+		// Retry 403 rate limits
+		if err, ok := err.(*downloadError); ok && err.code == 403 {
+			// Check if the error is a rate limit error
+			var message map[string]interface{}
+			jerr := json.Unmarshal(err.body, &message)
+			if jerr != nil {
+				return nil, -1, fmt.Errorf("cannot decode github response: %w", jerr)
+			}
+			if message["message"] != "API rate limit exceeded" {
+				return nil, -1, err
```

Comment:
We don't need to do a full map decode here. We can pull just the message field with an anonymous struct.

(Also, instead of jerr, if we inline the err into the if statement, the second name becomes unnecessary because the new `err` will exist only for the if statement.)

```suggestion
			var res struct{ Message string `json:"message"` }
			if err := json.Unmarshal(err.body, &res); err != nil {
				return nil, -1, fmt.Errorf("cannot decode github response: %w", err)
			}
			if res.Message != "API rate limit exceeded" {
				return nil, -1, err
```